lib/diagnostic.js

```diff
@@ -8,22 +8,38 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
-
+  if (typeof arguments[0] === 'undefined' ){
+    return undefined;
+  } else {
+    return Math.min.apply(null, arguments);
+  }
 };
```

Comment:
Your solution works exactly as expected. However, you could have accomplished this with simpler code. Your 1st if-clause isn't actually necessary. Take a look at this solution below:
```
const min = function min() {
  let result = arguments[0];
  for (let i = 1; i < arguments.length; i++) {
    if (arguments[i] < result) {
      result = arguments[i];
    }
  }

  return result;
};
```